src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -1405,25 +1511,19 @@ private void exportKey(APDU apdu) {
       len = publicKey.getW(apduBuffer, off);
       apduBuffer[(short) (off - 1)] = (byte) len;
       off += len;
-    } else if (publicOnly) {
+    } else {
       apduBuffer[off++] = TLV_PUB_KEY;
       off++;
       len = secp256k1.derivePublicKey(derivationOutput, (short) 0, apduBuffer, off);
       apduBuffer[(short) (off - 1)] = (byte) len;
       off += len;
     }
-
-    if (!publicOnly) {
-      apduBuffer[off++] = TLV_PRIV_KEY;
+    
+    if (withChaincode) {
+      apduBuffer[off++] = TLV_CHAIN_CODE;
       off++;
-
-      if (!derive || makeCurrent) {
-        len = privateKey.getS(apduBuffer, off);
-      } else {
-        Util.arrayCopyNonAtomic(derivationOutput, (short) 0, apduBuffer, off, Crypto.KEY_SECRET_SIZE);
-        len = Crypto.KEY_SECRET_SIZE;
-      }
-
+      Util.arrayCopyNonAtomic(chainCode, (short) 0, apduBuffer, off, CHAIN_CODE_SIZE);
```

Comment:
Same thing here. Not all that sensitive since these are just pubkeys, but I would still prefer atomic `arrayCopy`. Won't make it a requirement, but nice to have

Reply:
Per our conversation, I am not going to use atomic copies, unless saving sensitive data